### PR TITLE
Fix issue with ClassVar parsing

### LINKED ISSUE #3401: Pydantic fail to parse ClassVar annotation when it's a string
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
            pydantic version: 1.8.2
            pydantic compiled: False
                 install path: /Users/yuriikarabas/my-projects/pydantic/pydantic
               python version: 3.9.8 (tags/v3.9.8:bb3fdcfe95, Nov 10 2021, 14:54:52)  [Clang 13.0.0 (clang-1300.0.29.3)]
                     platform: macOS-12.0.1-x86_64-i386-64bit
     optional deps. installed: ['devtools', 'dotenv', 'email-validator', 'typing-extensions']
```

```py
from typing import ClassVar

from pydantic import BaseModel


class Model(BaseModel):
    val: "ClassVar[int]" = 10
```
```
Traceback (most recent call last):
  File "/Users/yuriikarabas/my-projects/pydantic/temp.py", line 6, in <module>
    class Model(BaseModel):
  File "/Users/yuriikarabas/my-projects/pydantic/pydantic/main.py", line 187, in __new__
    annotations = resolve_annotations(namespace.get('__annotations__', {}), namespace.get('__module__', None))
  File "/Users/yuriikarabas/my-projects/pydantic/pydantic/typing.py", line 340, in resolve_annotations
    value = _eval_type(value, base_globals, None)
  File "/Users/yuriikarabas/my-projects/cpython/Lib/typing.py", line 292, in _eval_type
    return t._evaluate(globalns, localns, recursive_guard)
  File "/Users/yuriikarabas/my-projects/cpython/Lib/typing.py", line 553, in _evaluate
    type_ = _type_check(
  File "/Users/yuriikarabas/my-projects/cpython/Lib/typing.py", line 158, in _type_check
    raise TypeError(f"{arg} is not valid as type argument")
TypeError: typing.ClassVar[int] is not valid as type argument
```

### PR DESCRIPTION
**Pull request by uriyyo**
## Related issue number
Fixes #3401

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by uriyyo** [comment]
please review

**Review by dusty-phillips — APPROVED** [review]

File: pydantic/typing.py [hunk 1]
```diff
@@ -332,7 +332,9 @@ def resolve_annotations(raw_annotations: Dict[str, Type[Any]], module_name: Opti
     annotations = {}
     for name, value in raw_annotations.items():
         if isinstance(value, str):
-            if sys.version_info >= (3, 7):
+            if (3, 10) > sys.version_info >= (3, 9, 8) or sys.version_info >= (3, 10, 1):
```
Comment (zekiblue) [inline pydantic/typing.py:335]: Is this include 3.10 ? isn't it better to do
```suggestion
            if sys.version_info >= (3, 9, 8):
```

Comment (uriyyo) [reply pydantic/typing.py:335]: This issue is not present in version `3.10.0` but presents in the current python `3.10` branch that will become python `3.10.1` on the next release.

For more context, this issue was introduced by this cpython PR - https://github.com/python/cpython/pull/28279

Comment (zekiblue) [reply pydantic/typing.py:335]: Thanks for the info!

Comment (uriyyo) [reply pydantic/typing.py:335]: You're welcome)

**Review by PrettyWood — APPROVED** [review]

**Comment by PrettyWood** [comment]
Thank you so much

**Pull request merged**